src/backends/perf_pt/collect.c

```diff
@@ -265,7 +265,22 @@ open_perf(pid_t target_tid) {
     attr.precise_ip = 3;
 
     // Acquire file descriptor through which to talk to Intel PT.
-    ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+    //
+    // This syscall could return EBUSY, meaning another process or thread has
+    // locked the Perf device. We spin around until we get an fd.
+    while (true) {
+        ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+        if (ret == -1) {
+            if (errno == EBUSY) {
+                continue;
+            } else {
+                ret = -1;
```

Comment:
`ret` is -1 by definition at this point so this assignment is unneeded?

Reply:
Correct, good spot.